src/AutoMapper/TypePair.cs

```diff
@@ -32,7 +30,7 @@ public override bool Equals(object obj)
             return obj is MapRequest && Equals((MapRequest) obj);
         }
 
-        public override int GetHashCode() => _hashcode;
+        public override int GetHashCode() => HashCodeCombiner.Combine(RequestedTypes, RuntimeTypes);
```

Comment:
That comment explains why and measuring confirms it. The main reason is reusing the code, but it's also slightly faster and uses less space.